source/initial_temperature/lithosphere_mask.cc

```diff
@@ -0,0 +1,195 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/initial_temperature/lithosphere_mask.h>
+#include <aspect/utilities.h>
+#include <limits>
+#include <aspect/geometry_model/interface.h>
+
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    LithosphereMask<dim>::LithosphereMask ()
+      :
+      lab_depths(1, 1.0)
+    {}
+
+
+    template <int dim>
+    void
+    LithosphereMask<dim>::initialize ()
+    {
+    if (LAB_depth_source == File)
+    {
+      const std::string filename = data_directory+LAB_file_name;
+
+      std::cout << "   Loading Ascii data lookup file " << filename << "." << std::endl;
+
+      lab_depths.load_file(filename,this->get_mpi_communicator());
+    }
+    }
+
+    //Read in ascii data - two dimensions so the third column is treated as data
+    template <int dim>
+    double
+    LithosphereMask<dim>::ascii_lab (const Point<2> &position) const
+    {
+      const double lab = lab_depths.get_data(position,0)*1000; //In km
```

Comment:
Either document in the class description at the bottom of this file that the file has to be in `km`, or (my preference) only allow files that input `m`, or (if you think it is important to support different units) add an input parameter that is a scale factor and is used here to multiply the value.